pkg/action/executor/resize_container.go

```diff
@@ -0,0 +1,287 @@
+package executor
+
+import (
+	kclient "k8s.io/client-go/kubernetes"
+	k8sapi "k8s.io/client-go/pkg/api/v1"
+
+	"github.com/turbonomic/kubeturbo/pkg/action/turboaction"
+	"github.com/turbonomic/kubeturbo/pkg/action/util"
+	"github.com/turbonomic/kubeturbo/pkg/discovery/monitoring/kubelet"
+	idutil "github.com/turbonomic/kubeturbo/pkg/discovery/util"
+	"github.com/turbonomic/kubeturbo/pkg/turbostore"
+	goutil "github.com/turbonomic/kubeturbo/pkg/util"
+	"github.com/turbonomic/turbo-go-sdk/pkg/proto"
+
+	"fmt"
+	"github.com/golang/glog"
+	"time"
+)
+
+type ContainerResizer struct {
+	kubeClient        *kclient.Clientset
+	kubeletClient     *kubelet.KubeletClient
+	broker            turbostore.Broker
+	k8sVersion        string
+	noneSchedulerName string
+
+	//a map for concurrent control of Actions
+	lockMap *util.ExpirationMap
+}
+
+func NewContainerResizer(client *kclient.Clientset, kubeletClient *kubelet.KubeletClient, broker turbostore.Broker, k8sver, noschedulerName string, lmap *util.ExpirationMap) *ContainerResizer {
+	return &ContainerResizer{
+		kubeClient:        client,
+		kubeletClient:     kubeletClient,
+		broker:            broker,
+		k8sVersion:        k8sver,
+		noneSchedulerName: noschedulerName,
+		lockMap:           lmap,
+	}
+}
+
+// get node cpu frequency, in KHz;
+func (r *ContainerResizer) getNodeCPUFrequency(host string) (uint64, error) {
+	return r.kubeletClient.GetMachineCpuFrequency(host)
+}
+
+func (r *ContainerResizer) setCPUCapacity(cpuMhz float64, host string, rlist k8sapi.ResourceList) error {
+	cpuFrequency, err := r.getNodeCPUFrequency(host)
```

Comment:
Do we need to call API every time we execute `setCPUCapacity`? I think this meta data can be store in a map and frequency of CPU is not likely to change so it is more likely a one time task.

Reply:
Yea,  right, node meta data should be cached.  I'd like to submit another PR to add cache for Node and pod.